### PR TITLE
Hiding the camp namespace

### PR DESCRIPTION
**Pull request by artv3**
@rhornung67 , I was wondering if this is what you had in mind?

### CODE REVIEW

File: include/RAJA/pattern/nested.hpp [hunk 1]
```diff
@@ -279,6 +278,8 @@ RAJA_INLINE void forall(const Pol &p, const SegmentTuple &st, const Body &b)
 
 }  // end namespace RAJA
 
+//hide namespace
```
[the remark this answers is not in the document]
Comment (ajkunen) [reply include/RAJA/pattern/nested.hpp]: I agree with @rhornung67 

Also, we could have an alias for the tuple class
namespace RAJA{
  using tuple = camp::tuple;
}

[the remark this answers is not in the document]
Comment (rhornung67) [reply include/RAJA/pattern/nested.hpp]: @ajkunen good idea.

File: include/RAJA/pattern/nested.hpp [hunk 2]
```diff
@@ -275,12 +275,13 @@ RAJA_INLINE void forall(const Pol &p, const SegmentTuple &st, const Body &b)
 
 }  // end namespace nested
 
+template<typename... Args>
+RAJA_HOST_DEVICE constexpr auto make_tuple(Args&&... args){
```
Comment (willkill07) [inline include/RAJA/pattern/nested.hpp]: auto without trailing return type requires c++14 support

you can add the trailing return type as follows (note: it's also important to perfectly forward any universal references):

```cpp
template<typename... Args>
RAJA_HOST_DEVICE constexpr auto make_tuple(Args&&... args)
-> decltype(camp::make_tuple(std::forward<Args>(args)...)) {
  return camp::make_tuple(std::forward<Args>(args)...);
}
```

Comment (willkill07) [reply include/RAJA/pattern/nested.hpp]: Alternatively, I believe adding `using ::camp::make_tuple;` within the RAJA namespace should enable visibility of `RAJA::make_tuple` without the need for a redefinition

**Comment by artv3** [comment]
Hey... I got a weird error from Travis: 
ERROR: An error occured while trying to fetch your .travis.yml file.

**Comment by rhornung67** [comment]
@artv3 I just restarted the build. Let's see what happens.

**Comment by rhornung67** [comment]
@artv3 After restarting, all travis checks passed. Not sure why that doesn't show up here in github.

**Comment by rhornung67** [comment]
@artv3 try merging develop into this branch. @davidbeckingsale just merged a change that should fix the appveyor build.  After it passes, I'll approve this.

**Comment by artv3** [comment]
@rhornung67, thanks! I am actually not yet authorized to merge the branch.

**Comment by rhornung67** [comment]
You have write privileges for the project so you should be able to merge. Are you logged in to github?

**Pull request merged**